### PR TITLE
Ryan eric karab

### PR DESCRIPTION
**Pull request by aesuan**

### CODE REVIEW

File: src/components/ChatRoom.jsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React, { useState, useEffect } from "react";
 import SocketEvents from "../lib/enums/socketEvents";
 import moment from "moment";
 import Chat from "./Chat";
+import GameLobby from "../lib/gameLogic/GameLobby.js";
```
Comment (aesuan) [inline src/components/ChatRoom.jsx:5]: Is this being used in this file?

File: src/components/SetUser.jsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import UserList from "./UserList";
 
 function SetUser({ socket, setUserName, userName }) {
   const [userNameInput, updateUserName] = useState("");
-  const [readyStatus, setReadySatus] = useState(false);
+  const [readyStatus, setReadyStatus] = useState(false);
```
Comment (aesuan) [inline src/components/SetUser.jsx:7]: nice catch lol

File: src/lib/gameLogic/GameLobby.js [hunk 3]
```diff
@@ -221,6 +224,24 @@ class GameLobby {
...
+    // console.log("MESSAGE FROM GAMELOBBY : ", this.messages);
+    let chatMessage = '';
+    if (message.evaluate) {
+      chatMessage = `${message.userName} guessed the word.`;
+    } else {
+      chatMessage = `${message.userName}: ${message.guess}`;
+    }
+    const newChatMessage = new EventOptions({
+      message: {message: chatMessage},
+      event: SocketEvents.USER_MESSAGE,
```
Comment (aesuan) [inline src/lib/gameLogic/GameLobby.js]: think this should be changed to Lobby message which is set up on the front end already to print messages to chat!

File: src/components/ChatRoom.jsx [hunk 4]
```diff
@@ -38,10 +41,13 @@ function ChatRoom({ socket, userName }) {
       <input
         type="submit"
         value="enter"
-        onClick={() => submitChatMessage({ guess: chatInput, userName })}
+        onClick={() => {
+          submitChatMessage({ guess: chatInput, userName });
+          updateChatInput('');
```
Comment (aesuan) [inline src/components/ChatRoom.jsx:45]: good call clearing the input!

File: src/components/ChatRoom.jsx [hunk 5]
```diff
@@ -10,18 +11,20 @@ function ChatRoom({ socket, userName }) {
   useEffect(() => {
     function addMessages(message) {
       let newMessages = messages.concat(message);
+      console.log('USE EFFECT newMessages : ',newMessages);
       updateMessage(newMessages);
     }
     socket.on(SocketEvents.LOBBY_MESSAGE, addMessages);
+    socket.on(SocketEvents.USER_MESSAGE, addMessages);
```
Comment (aesuan) [inline src/components/ChatRoom.jsx]: probably don't need to add user message here! would just use the lobby message event already there. User message should be to just one user.

File: src/components/Chat.jsx [hunk 6]
```diff
@@ -2,9 +2,15 @@ import React from "react";
 import moment from "moment";
 
 function Chat({ message }) {
+  console.log('CHAT.JSX - MESSAGE : ', message)
+  if (typeof message.message === 'object') {
```
Comment (aesuan) [inline src/components/Chat.jsx]: is this necessary? seems like the message is g=being processed on the backend to be in the message property

File: server/socket/index.js [hunk 7]
```diff
@@ -28,7 +29,14 @@ dynamicNspLobby.on("connect", (socket) => {
 
   socket.on(SocketEvents.SUBMIT_CHAT_MESSAGE, (data, callback) => {
     let isCorrect = currentLobby.evaluateGuess(data);
-    callback(isCorrect);
```
Comment (aesuan) [inline server/socket/index.js:31]: callback can be useful for the user sending the original message to get special info if you needed, but seems like not necessary for now!

File: src/lib/gameLogic/GameLobby.js [hunk 8]
```diff
@@ -221,6 +224,24 @@ class GameLobby {
...
   }
 
+  updateMessage(message) {
+    this.messages.push(message);
+    console.log('message:', message);
+    // console.log("MESSAGE FROM GAMELOBBY : ", this.messages);
+    let chatMessage = '';
+    if (message.evaluate) {
+      chatMessage = `${message.userName} guessed the word.`;
+    } else {
```
Comment (aesuan) [inline src/lib/gameLogic/GameLobby.js:234]: might want to think about if in the future you would want to separate username from message for styling and process that on the front end but think this makes sense for now! also not sure if you want the message to appear differently if is correct or not ie green text, and therefore would want a success boolean or some other property

**Review by aesuan — COMMENTED** [review]
the transpiled files (ie in public folder) should be gitignored in the future! looks good with one quick adjustment

**Review by ucb15erickao — APPROVED** [review]

**Pull request merged**